Close button moved outside of tab element

Blackbaud-AlexKingman opened the pull request:
#88

blackbaud-ado commented:
Preview build at [https://developer.blackbaud.com/skyux-tabs?_sp.spa=pr-close-button-a11y-fix](https://developer.blackbaud.com/skyux-tabs?_sp.spa=pr-close-button-a11y-fix)

blackbaud-ado commented:
Preview build at [https://developer.blackbaud.com/skyux-tabs?_sp.spa=pr-close-button-a11y-fix](https://developer.blackbaud.com/skyux-tabs?_sp.spa=pr-close-button-a11y-fix)

johnhwhite commented on src/app/public/modules/tabs/tab-button.component.scss:
```diff
@@ -2,23 +2,59 @@
+@include sky-theme-modern-dark {
+  .sky-btn-tab-wrapper {
+    .sky-btn-tab-close {
+      color: #eeeeef;
```
This should use `$sky-background-color-neutral-light`

Blackbaud-AlexKingman replied:
Good catch. Done!

johnhwhite commented on src/app/public/modules/tabs/tab-button.component.html:
```diff
@@ -1,34 +1,40 @@
 >
-  <span
-    class="sky-tab-heading"
+    <a
```
Tag should be indented two spaces.

Blackbaud-AlexKingman replied:
Fixed.

johnhwhite commented on src/app/public/modules/tabs/tabset.component.scss:
```diff
@@ -77,9 +80,15 @@
+
+  .sky-tab-dropdown-menu-item {
+    .sky-btn-tab-wrapper {
+      margin-right: 0 !important;
```
I think this can be done without the `!important` here.

Blackbaud-AlexKingman replied:
Yep, just had to remove encapsulation on the tab-button component.

johnhwhite commented on src/app/public/modules/tabs/tabset.component.scss:
```diff
@@ -45,24 +45,27 @@
-      display: flex;
+  .sky-btn-tab-wrapper {
+    width: 100%;
+    margin-right: 0 !important;
```
I think this can be done without `!important`.

Blackbaud-AlexKingman replied:
Yep, just had to remove encapsulation on the tab-button component.

blackbaud-ado commented:
Preview build at [https://developer.blackbaud.com/skyux-tabs?_sp.spa=pr-close-button-a11y-fix](https://developer.blackbaud.com/skyux-tabs?_sp.spa=pr-close-button-a11y-fix)

Blackbaud-MattGregg commented:
@Blackbaud-AlexKingman Looks like the close button moving out of tab element is good. Couple things I noticed still could use some attention:
1. The close button label. They all have a generic label and not specific calling out the title of the tab to give more context (make meaningful as was shown in recommendation). Can the text label of the button be inserted into the aria-label of the close button "Close [tab label] tab"
2. Did you try this in Voiceover at all. I wasn't getting the close button to actually work and close the tab. Nothing would happen when I selected it via Voiceover. I could use/activate the tabs just fine.
3.

blackbaud-ado commented:
Preview build at [https://developer.blackbaud.com/skyux-tabs?_sp.spa=pr-close-button-a11y-fix](https://developer.blackbaud.com/skyux-tabs?_sp.spa=pr-close-button-a11y-fix)

Blackbaud-AlexKingman commented:
> 1. The close button label. They all have a generic label and not specific calling out the title of the tab to give more context (make meaningful as was shown in recommendation). Can the text label of the button be inserted into the aria-label of the close button "Close [tab label] tab"
> 2. Did you try this in Voiceover at all. I wasn't getting the close button to actually work and close the tab. Nothing would happen when I selected it via Voiceover. I could use/activate the tabs just fine.

@Blackbaud-MattGregg 
1. Good point. I've updated the code to grab the tab string and wedge it into the label as you suggest. However, we're having some trouble with the locale resource strings right now (which controls any copy inside the tab button), so you won't see that update in this PR until the package merges... In the meantime, you can see the code I updated to support this [here](https://github.com/blackbaud/skyux-tabs/pull/162/files#diff-f7888e279395499cedc890b0e93a5709edc750524622091ad3beecf349d5970cR8)
2. That's just an artifact of the visual test - it has nothing to do with Voiceover. We haven't wired up this particular visual test with all the bells and whistles yet, so clicking the close button does nothing. You can, however, see the close functionality working in the tabs demo page under this PR.

Blackbaud-ToddRoberts reviewed (commented):
Width of the tab appears to be different compared to the existing component demo.

blackbaud-ado commented:
Preview build at [https://developer.blackbaud.com/skyux-tabs?_sp.spa=pr-close-button-a11y-fix](https://developer.blackbaud.com/skyux-tabs?_sp.spa=pr-close-button-a11y-fix)

Blackbaud-MattGregg reviewed (approved):
@Blackbaud-AlexKingman Gotcha on the tab label strings coming in later. I see that in the code and I'm sure you're on it. I did test out the demo keyboard and voiceover, close buttons work. thx.
